wpilibj/src/shared/java/edu/wpi/first/wpilibj/PIDController.java

```diff
@@ -284,7 +287,9 @@ protected void calculate() {
                 + calculateFeedForward();
           }
         } else {
-          if (m_I != 0) {
+          if (Math.abs(m_error) > m_maxErrorToIntegrate) {
+            m_totalError = 0.0;
+          } else if (m_I != 0) {
```

Comment:
Since the accumulator error is reset outside of a certain range, it doesn't make sense to then start accumulating again without sampling the error again in between.